packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -454,127 +461,296 @@ export class ModularChangeFamily
+							baseChanges,
+							fieldFilter,
 							revisionMetadata,
-						),
+						);
+					},
 					genId,
 					manager,
 					revisionMetadata,
 				);
-
 				const rebasedFieldChange: FieldChange = {
 					fieldKind: fieldKind.identifier,
-					change: brand(rebasedField),
-				};
-
-				const rebaseData: RebaseData = {
-					fieldChange: rebasedFieldChange,
-					baseChange: taggedBaseChange,
+					change: brand(rebasedChangeset),
 				};
-
-				addFieldData(manager, rebaseData);
 				rebasedFields.set(field, rebasedFieldChange);
+				addFieldData(manager, rebasedChangeset);
 			}
 		}
 
 		return rebasedFields;
 	}
 
 	private rebaseNodeChange(
-		change: NodeChangeset,
-		over: TaggedChange<NodeChangeset>,
+		change: NodeChangeset | undefined,
+		over: TaggedChange<NodeChangeset> | undefined,
 		genId: IdAllocator,
-		crossFieldTable: CrossFieldTable<RebaseData>,
+		crossFieldTable: RebaseTable,
+		parentField: FieldChange | undefined,
+		fieldFilter: (baseChange: FieldChange, newChange: FieldChange | undefined) => boolean,
 		revisionMetadata: RevisionMetadataSource,
-	): NodeChangeset {
-		if (change.fieldChanges === undefined || over.change.fieldChanges === undefined) {
-			return change;
-		}
-
-		return {
-			...change,
-			fieldChanges: this.rebaseFieldMap(
-				change.fieldChanges,
-				{
-					...over,
-					change: over.change.fieldChanges,
-				},
-				genId,
-				crossFieldTable,
-				revisionMetadata,
-			),
-		};
+	): NodeChangeset | undefined {
+		if (change === undefined && over?.change?.fieldChanges === undefined) {
+			return undefined;
+		}
+
+		if (over?.change?.fieldChanges !== undefined && parentField !== undefined) {
+			crossFieldTable.baseMapToParentField.set(over.change.fieldChanges, parentField);
+		}
+
+		const baseMap: TaggedChange<FieldChangeMap> =
+			over?.change?.fieldChanges !== undefined
+				? {
+						...over,
+						change: over.change.fieldChanges,
+				  }
+				: makeAnonChange(new Map());
+
+		const fieldChanges = this.rebaseFieldMap(
+			change?.fieldChanges ?? new Map(),
+			baseMap,
+			genId,
+			crossFieldTable,
+			fieldFilter,
+			revisionMetadata,
+		);
+
+		if (change?.valueChange === undefined && fieldChanges.size === 0) {
+			return undefined;
+		}
+
+		const rebasedChange: NodeChangeset = {};
+		if (change?.valueChange !== undefined) {
+			rebasedChange.valueChange = change.valueChange;
+		}
+
+		if (fieldChanges.size > 0) {
+			rebasedChange.fieldChanges = fieldChanges;
+		}
+
+		if (over?.change?.fieldChanges !== undefined) {
+			crossFieldTable.baseMapToRebased.set(over.change.fieldChanges, rebasedChange);
+		}
+
+		return rebasedChange;
 	}
 
 	public rebaseAnchors(anchors: AnchorSet, over: ModularChangeset): void {
 		anchors.applyDelta(this.intoDelta(over));
 	}
 
-	public intoDelta(change: ModularChangeset): Delta.Root {
-		return this.intoDeltaImpl(change.changes);
+	public intoDelta(change: ModularChangeset, repairStore?: ReadonlyRepairDataStore): Delta.Root {
```

Comment:
Some leftover from merging Jenn's changes I'm guessing
```suggestion
	public intoDelta(change: ModularChangeset): Delta.Root {
```